pgrx-sql-entity-graph/src/used_type.rs

```diff
@@ -608,6 +613,146 @@ fn resolve_option_inner(
     }
 }
 
+fn resolve_result_inner(
+    original: syn::TypePath,
+) -> syn::Result<(syn::Type, Option<CompositeTypeMacro>)> {
+    let segments = &original.path;
+    let last = segments
+        .segments
+        .last()
+        .ok_or(syn::Error::new(original.span(), "Could not read last segment of path"))?;
+
+    // Get the path of our Result type, to handle crate::module::Result pattern
+    // TODO - more robust / idiomatic way of handling this?
+    let mut without_type_args = original.path.clone();
+    without_type_args.segments.last_mut().unwrap().arguments = syn::PathArguments::None;
+ 
+    let (ok_ty, err_ty) = { 
+        if let syn::PathArguments::AngleBracketed(path_arg) = &last.arguments { 
+            if let Some(first_ty) = path_arg.args.first() {
+                // Since `pub type Result<T> = std::error::Result<T, OurError>
+                // is a common pattern, 
+                // we should support single-argument Result<T> style.
+                if path_arg.args.len() == 1 {
+                    (first_ty.clone(), None)
+                }
+                else if path_arg.args.len() == 2 { 
+                    (first_ty.clone(), Some(path_arg.args[1].clone()))
+                }
+                else { 
+                    return Err(syn::Error::new(
+                        last.arguments.span(),
+                        "Result<..> with more than two type arguments not supported.",
+                    ));
+                }
+            }
+            else { 
+                // Return early, Result<> with no type args.
+                return Err(syn::Error::new(
+                    last.arguments.span(),
+                    "Cannot return a Result without type generic arguments.",
+                ));
+            }
```

Comment:
Can this just be
```rust
let mut iter = args.iter();
match (iter.next(), iter.next(), iter.next()) {
```

So that we can have the compiler check that we're exhaustively handling all cases we want to handle, and we don't have the problem of trying to track whether the indexing we do later is correct? Seeing `.first()` and then `first_ty` and then `[1]` is lexically confusing.

Reply:
Sadly, `syn` doesn't expose a slice type here, so we can't just
```rust
match &slice_of_generic_args {
    [a, b, c, ..] => return Err(..),
    [a, b] => { /* code */ },
    [a] => { /* code */ },
    [] => return Err(..),
}
```

Reply:
This should be cleaner now, let me know if the new implementation fits.